liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -1563,6 +1571,89 @@ public void execute() throws MojoExecutionException {
 
         doDevMode();
     }
+    
+    protected List<MavenProject> getMultiModuleDevModeProjects(ProjectDependencyGraph graph) throws MojoExecutionException {
+    	getLog().debug("Resolve dev mode projects");
+    	
+        List<MavenProject> sortedReactorProjects = graph.getSortedProjects();
+        Set<MavenProject> conflicts = new LinkedHashSet<MavenProject>(); // keeps the order of items added in
+
+        // A leaf here is a module without any downstream modules depending on it
+        List<MavenProject> leaves = new ArrayList<MavenProject>();
+        for (MavenProject reactorProject : sortedReactorProjects) {
+            if (graph.getDownstreamProjects(reactorProject, true).isEmpty()) {
+                leaves.add(reactorProject);
+                getLog().debug("Found final downstream project: " + reactorProject.getArtifactId());
+            }
+        }
+        
+        // Remove jar projects since these are not relavent to dev mode
+        List<MavenProject> jarLeaves = new ArrayList<MavenProject>();
+        for (MavenProject leaf : leaves) {
+        	if (leaf.getPackaging().equals("jar")) {
+        		jarLeaves.add(leaf);
+        		getLog().debug("Removing jar project: " + leaf.getArtifactId());
+        	}
+        }
+        leaves.removeAll(jarLeaves);
+        
+        // Remove any projects that are configured to be skipped
+        List<MavenProject> skipLeaves = new ArrayList<MavenProject>();
+        for (MavenProject leaf : leaves) {
+        	
+            // Properties that are set in the pom file
+            Properties props = leaf.getProperties();
+
+            // Properties that are set by user via CLI parameters
+            Properties userProps = session.getUserProperties();
+        	Plugin libertyPlugin = getLibertyPluginForProject(leaf);
+            Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(libertyPlugin, "dev", getLog());
+            
+            boolean skipGoal = DevHelper.getBooleanFlag(config, userProps, props, "skip");
+            
+        	if (skipGoal) {
+        		getLog().debug("Skip configured on project: " + leaf.getArtifactId());
+        		skipLeaves.add(leaf);
```

Comment:
This also seems like something that could be added to original `for` loop. If you want to put the check in a separate method, that could make the original `for` loop not get too long and cluttered. But I see no reason for three separate `for` loops...if ultimately you only want `leaves` to contain leaves that are not `jar` and do not have `skip` configured.

Reply:
Good point.